### PR TITLE
Remove PubKeyHash to PubKeyAddress map

### PR DESCRIPTION
**Pull request by nazrhom**
Change-type: patch
Signed-off-by: Giovanni Garufi <giovanni@mlabs.city>

### CODE REVIEW

File: src/TokenAirdrop.hs [hunk 1]
```diff
@@ -113,3 +108,11 @@ group :: Int -> [a] -> [[a]]
 group n list
   | length list <= n = [list]
   | otherwise = let (xs, xss) = splitAt n list in xs : group n xss
+
```
Comment (nazrhom) [inline src/TokenAirdrop.hs:111]: I copied this over from another project, we have a version of this in plutus-extra we could use. Wasn't sure if it is worth adding it in here or as part of a separate PR

Comment (samuelWilliams99) [reply src/TokenAirdrop.hs:111]: Gonna add this as a comment to #26 so we can handle it there.

File: src/BeneficiariesFile.hs [hunk 2]
```diff
@@ -83,17 +84,18 @@ parseAssetOrAmt str = (Left <$> parseAsset str) <> (Right <$> parseAmt str)
...
-      toPubKeyAddress' $ pubKeyHashAddress pkh
+      toAddress $ pubKeyHashAddress pkh
     else do
       addr <- deserialiseAddress addrStr
-      toPubKeyAddress' addr
+      toAddress addr
   where
-    toPubKeyAddress' = maybeToRight ("Script addresses are not allowed: " <> addrStr) . toPubKeyAddress
+    toAddress addr@(Address (PubKeyCredential _) _) = Right addr
+    toAddress _ = Left $ "Script addresses are not allowed: " <> addrStr
```
Comment (samuelWilliams99) [inline src/BeneficiariesFile.hs:98]: #28 
Just an FYI, no changes needed here

File: src/FakePAB/Address.hs [hunk 3]
```diff
@@ -71,22 +68,3 @@ deserialiseAddress' eraType addr = do
...
-
-toPubKeyAddress :: Address -> Maybe PubKeyAddress
-toPubKeyAddress (Address cred stakingCred) =
-  case cred of
-    PubKeyCredential pkh -> Just $ PubKeyAddress pkh stakingCred
-    ScriptCredential _ -> Nothing
-
-fromPubKeyAddress :: PubKeyAddress -> Address
-fromPubKeyAddress (PubKeyAddress pkh stakingCred) =
-  Address (PubKeyCredential pkh) stakingCred
```
Comment (samuelWilliams99) [inline src/FakePAB/Address.hs:92]: Love to see relic code going away :)

File: src/TokenAirdrop.hs [hunk 4]
```diff
@@ -32,28 +31,24 @@ tokenAirdrop config = do
             map
               ( \beneficiary ->
                   let val = Value.assetClassValue beneficiary.assetClass beneficiary.amount
-                   in (Constraints.mustPayToPubKey (PaymentPubKeyHash beneficiary.address.pkaPubKeyHash) val, [beneficiary])
+                   in (mustPayToAddress (beneficiary.address) val, [beneficiary])
```
Comment (samuelWilliams99) [inline src/TokenAirdrop.hs:34]:
```suggestion
                   in (mustPayToAddress beneficiary.address val, [beneficiary])
```
Record dot brackets for you :)

**Review by samuelWilliams99 — APPROVED** [review]

**Pull request merged**